Extension/src/LanguageServer/configurations.ts

```diff
@@ -628,6 +628,27 @@ export class CppProperties {
             configuration.cStandard = this.updateConfigurationString(configuration.cStandard, settings.defaultCStandard, env);
             configuration.cppStandard = this.updateConfigurationString(configuration.cppStandard, settings.defaultCppStandard, env);
             configuration.intelliSenseMode = this.updateConfigurationString(configuration.intelliSenseMode, settings.defaultIntelliSenseMode, env);
+            if ((configuration.compilerPath === undefined || configuration.compilerPath === "") && !!this.defaultCompilerPath) {
+                configuration.compilerPath = this.defaultCompilerPath;
+                if ((!configuration.cStandard || configuration.cStandard === "") && !!this.defaultCStandard) {
```

Comment:
Is it intentional to condition these on the compilerPath?

Reply:
Yes.  If we're not actually using the detected default compiler, we should not apply the detected default compiler's standards, intelliSenseMode, etc.  Right?

Reply:
But, I'm not sure about defaultWindowsSdkVersion, defaultIncludes, defaultFrameworks .